### DISCUSSION TITLE
Hii guys, I am stuck with this null properties issue , the return props giving NULL but when I tried with custom ObjectId instead of  ctx.params.postId  it worked

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#58801] [App Router] [answered]
![Screenshot from 2023-11-23 07-52-03](https://github.com/vercel/next.js/assets/113231164/d9a52649-cd39-40cd-ab38-3d9de11bb8b8)

![Screenshot from 2023-11-23 07-52-28](https://github.com/vercel/next.js/assets/113231164/4afdeb9a-13bf-4554-b582-93679d05a573)

import { getSession, withPageAuthRequired } from "@auth0/nextjs-auth0";
import { AppLayout } from "../../components/AppLayout";
import clientPromise from "../../lib/mongodb.js";
import { ObjectId } from "mongodb";
import { redirect } from "next/dist/server/api-utils";
import { getAppProps } from "../../utils/getAppProps";

export default function Post(props) {
    console.log('PROPS: ',props);
    

    return <div className="overflow-auto h-full bg-slate-300">
      <div className="max-w-screen-sm mx-auto">
      <div className="text-5xl font-bold mt-6 p-2 bg-gray-500 rounded-sm">
        Keywords
       </div>
       <div className="flex flex-wrap pt-2 gap-1">
        {props.keywords.split(",").map((keyword, i) => (
          <div key={i} className="p-2 rounded-full bg-slate-800 text-white">
            {keyword}
          </div>
        ))}
      </div>

       <div className="text-5xl font-bold mt-6 p-2 bg-gray-500 rounded-sm">
        Blog
       </div>
       <div dangerouslySetInnerHTML={{__html: props.postContent || ''}}/>
      </div>
    </div>
  }
  

  Post.getLayout = function getLayout(page, pageProps) {
    return <AppLayout {...pageProps}>{page}</AppLayout>;
  };
  

  export const getServerSideProps = withPageAuthRequired ({
    async getServerSideProps(ctx){
      const props = await getAppProps(ctx);
      const userSession = await getSession(ctx.req, ctx.res);
      const client = await clientPromise;
      const db = client.db("BlogBraniac");
      const user = await db.collection("users").findOne({
        auth0Id : userSession.user.sub,
      });
      const post = await db.collection("posts").findOne({
        _id : new ObjectId(ctx.params.postId),
        userId: user._id,
      });

      
          

      return{
        props: {
          postContent: post.postContent,
          title: post.title,
          metaDescription: post.metaDescription,
          keywords: post.keywords,
          ...props,
      },
      };

    },
  });

**Comment by icyJoseph** [comment]
Your route is called `[...postid]`, so it should be `ctx.params.postid` (the `i` lowercased): 

## Edit

I forgot to mention that `[...postid]` resolves to a list, so we'd need to index the first item, or change the file name to `[postid]` - I got carried away with the lowercased issue.

```js
_id : new ObjectId(ctx.params.postid[0])
```

**Reply by ghost** [reply]
Hii joseph I tried that in morning but getting this below error:

![Screenshot from 2023-11-23 12-27-31](https://github.com/vercel/next.js/assets/113231164/c2614a86-cdd1-4076-ada1-d483e95327f9)

![Screenshot from 2023-11-23 15-50-05](https://github.com/vercel/next.js/assets/113231164/f604a755-9475-40a0-acb3-a046242d1043)


I also tried putting direct id's from mongoDB database and It worked there, I think ctx.params.postid  syntax not working

**Reply by icyJoseph** [reply, marked as answer]
Hi,

Yeah because you have `[...postId]`, and that means that `ctx.params.postId` will be a list. 

- https://nextjs.org/docs/pages/building-your-application/routing/dynamic-routes#catch-all-segments

I forgot to mention that as well.

You went from `undefined` to `["something"]`, just do `ctx.params.postId[0]`, or rename the file to `[postId]` 

### Yet another edit

It is good practice to learn to debug these things, there's no magic going on. `console.log(typeof ctx.params.postId)` or even `console.log(ctx.params)` would've helped you in this case. Also reading up the `new ObjectId()` documentation, and seeing that it can take no arguments, `undefined` or a string, and that's why it started to fail when the casing was fixed.

**Reply by ghost** [reply]
You saved the day man, thanks a lot Sir. I am smiling after so many hours. I did change the file name, but it was returning a BSON error which was new to me but the above solution worked. From now on would console log everything. Thanks again.

**Reply by icyJoseph** [reply]
Hehe no worries! Please don't forget to mark as answered. 

More than console log, in general learn debugging tools :) it takes time though.

Have a good day!
